stdio_lim.h not available in musl libc

## Description

nonstandard/interal gcc libc header `<bits/stdio_lim.h>` is used inside [`src/arch/posix/eventloop_posix.h`](https://github.com/open62541/open62541/blob/0c01de61b738daedac23293a3b10ff558813859c/arch/posix/eventloop_posix.h#L280), resulting in compilation failure when building with MUSL.

```
[ 23%] Building C object CMakeFiles/open62541-plugins.dir/arch/posix/eventloop_posix.c.o
In file included from /open62541/arch/posix/eventloop_posix.c:9:
/open62541/arch/posix/eventloop_posix.h:280:10: fatal error: bits/stdio_lim.h: No such file or directory
  280 | #include <bits/stdio_lim.h>
      |          ^~~~~~~~~~~~~~~~~~
compilation terminated.
```

I might be missing something, but it looks like just removing the include fixes the issue and dose not break any tests for glibc build.

## Background Information / Reproduction Steps

docker image used for testing:

```dockerfile
FROM alpine:3.23

RUN apk add --no-cache \
    cmake \
    make \
    gcc \
    musl-dev \
    python3 \
    check-dev \
    libpcap-dev \
    linux-headers \
    subunit-dev
```

Used CMake options:

```bash
cmake -DUA_BUILD_UNIT_TESTS=ON -DCMAKE_BUILD_TYPE=Release 
```

## Checklist

Please provide the following information:

 - [x] open62541 Version (release number or git tag): 0c01de61b738daedac23293a3b10ff558813859c
 - [ ] Other OPC UA SDKs used (client or server): 
 - [ ] Operating system:
 - [ ] Logs (with `UA_LOGLEVEL` set as low as necessary) attached
 - [ ] Wireshark network dump attached
 - [ ] Self-contained code example attached
 - [ ] Critical issue

--- a/arch/posix/eventloop_posix.h
+++ b/arch/posix/eventloop_posix.h
@@ -275,12 +275,6 @@ typedef int SOCKET;
 #endif /* !__APPLE__ */
 #include <sys/stat.h>
 
-#ifndef __ANDROID__
-#ifndef __APPLE__
-#include <bits/stdio_lim.h>
-#endif /* !__APPLE__ */
-#endif /* !__ANDROID__ */
-
 #define UA_STAT stat
 #define UA_DIR DIR
 #define UA_DIRENT dirent

